controllers/control-plane/kyma_remote_sync_test.go

```diff
@@ -243,6 +235,58 @@ var _ = Describe("Kyma sync into Remote Cluster", Ordered, func() {
 	})
 })
 
+var _ = Describe("Kyma sync default module list into Remote Cluster", Ordered, func() {
+	kyma := NewTestKyma("kyma")
+	moduleInKCP := NewTestModule("in-kcp", v1beta2.DefaultChannel)
+	kyma.Spec.Modules = []v1beta2.Module{{Name: moduleInKCP.Name, Channel: moduleInKCP.Channel}}
+
+	var runtimeClient client.Client
+	var runtimeEnv *envtest.Environment
+
+	remoteKyma := &v1beta2.Kyma{}
+	remoteKyma.Name = v1beta2.DefaultRemoteKymaName
+	remoteKyma.Namespace = controllers.DefaultRemoteSyncNamespace
+
+	BeforeAll(func() {
+		runtimeClient, runtimeEnv = NewSKRCluster(controlPlaneClient.Scheme())
+	})
+	registerControlPlaneLifecycleForKyma(kyma)
+
+	It("Kyma CR should be synchronized in both clusters", func() {
+		By("Remote Kyma created")
+		Eventually(kymaExists, Timeout, Interval).
+			WithArguments(runtimeClient, remoteKyma.Name, remoteKyma.Namespace).
+			Should(Succeed())
+
+		By("Remote Kyma contains default module")
+		Eventually(containsModuleInSpec, Timeout, Interval).
+			WithArguments(runtimeClient, remoteKyma.Name, remoteKyma.Namespace, moduleInKCP.Name).
+			Should(Succeed())
+
+		By("KCP Manifest is being created")
+		Eventually(ManifestExists, Timeout, Interval).
+			WithArguments(ctx, kyma, moduleInKCP, controlPlaneClient).
+			Should(Succeed())
+		By("KCP Manifest CR becomes ready")
+		Eventually(UpdateManifestState, Timeout, Interval).
+			WithArguments(ctx, controlPlaneClient, kyma, moduleInKCP, v1beta2.StateReady).
+			Should(Succeed())
+
+		By("Delete default module from remote Kyma")
+		Eventually(removeModuleFromKyma, Timeout, Interval).
+			WithArguments(runtimeClient, remoteKyma.Name, remoteKyma.Namespace, moduleInKCP.Name).
+			Should(Succeed())
+
+		By("KCP Manifest is being deleted")
```

Comment:
Please add a test case to check KCP Kyma still contains those initiated modules in spec, We need to make sure those init modules do not get removed.

The follow up test case can be
1. delete SKR Kyma CR
2. verify all KCP Manifests created again

Reply:
Done